lib/acts_as_rdfable/acts_as_rdfable_core.rb

```diff
@@ -31,6 +35,10 @@ def acts_as_rdfable(formats: [])
         RdfAnnotation.for_table_column(self.table_name, attr)
       end
 
+      define_singleton_method :attribute_for_rdf_annotation do |annotation|
+        RdfAnnotation.find_by(table: self.table_name, predicate: annotation)&.column
```

Comment:
Suggest moving the lookup logic into a scope in RdfAnnotation, the way the others are (probably a `for_table_predicate` scope that leverages the existing `for_table` scope the way `for_table_column` does?)

Reply:
The general principle here is that things outside the Model shouldn't hardcode column names about the model (information hiding) so that future table refactorings don't have to touch a bunch of different things, and scopes are an easy way to create little methods that hide the details of the lookup

Reply:
Made that change.  Is the main advantage with using scopes here chaining?

Reply:
I'd say, probably in roughly this order, the advantages of using scopes are:

1) Encapsulation/information hiding - putting the logic for the lookup in the model it belongs to rather than elsewhere. Hiding details from external callers is almost always the right move, far beyond scopes.

2) Graceful nil handling (the non-obvious part of chaining): https://www.justinweiss.com/articles/should-you-use-scopes-or-class-methods/

3) joins on associations work gracefully with scopes